### PR TITLE
Fix ISEVEN, ISODD and LEN

### PR DESCRIPTION
**Pull request by nikazooz**
- ISEVEN and ISODD should return error if provided value is not numeric.
- Since LEN handles arrays, I've added handling of empty array. Also, having `null` as (blank) input should return 0.
- Added some assertions for ISTEXT and ISLOGICAL

Let me know what you think about this.

Cheers!

### CODE REVIEW

**Comment by nicolashefti** [comment]
@nikazooz I am just wondering if the current implementation of LEN is correct. Normally it should only accept a String as argument. All the rest should trigger an error.value.

https://exceljet.net/excel-functions/excel-len-function

File: test/information.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ describe('Information', function() {
     information.ISEVEN(2.5).should.equal(true);
     information.ISEVEN(5).should.equal(false);
     information.ISEVEN(0).should.equal(true);
+    information.ISEVEN(null).should.equal(error.error);
```
Comment (nicolashefti) [inline test/information.js:62]: It is missing a bit of consistence between ISEVEN, ISODD, ISLOGICAL on the null case. I would always return a boolean value in all 3 functions.

Comment (nikazooz) [reply test/information.js:62]: In that case, what I did is incorrect

**Comment by nikazooz** [comment]
@nicolashefti For LEN, the main reason is that when a reference to empty cell is used LEN returns 0. One could argue that empty cell still has an empty string but to me it made sense for it to handle `null` the same way. Does it make sense?

File: test/information.js (hunk 1 above)
Comment (nicolashefti) [reply test/information.js:62]: Can you update to return "false" in case of "null" for the 3 functions ?

File: test/text.js [hunk 2]
```diff
@@ -91,6 +91,8 @@ describe('Text', function() {
     text.LEN('four').should.equal(4);
     text.LEN([1, 2, 3, 4, 5]).should.equal(5);
     text.LEN().should.equal(error.error);
+    text.LEN(null).should.equal(0);
+    text.LEN([]).should.equal(0);
```
Comment (nicolashefti) [inline test/text.js:95]: Can you remove the Array support as input ? I think this is a mistake of the early implementation.

text.LEN([1, 2, 3, 4, 5]).should.equal(error.error);
text.LEN([]).should.equal(error.error);

About edge cases, I would go for:

text.LEN('').should.equal(0);
text.LEN(null).should.equal(0);

**Review by nicolashefti — CHANGES_REQUESTED** [review]

File: test/information.js (hunk 1 above)
Comment (nikazooz) [reply test/information.js:62]: Should `ISEVEN` return `false` or `true? Currently, it returns true and Google Spreadsheets do the same. Don't know about Excel

Comment (nicolashefti) [reply test/information.js]: Indeed, ISEVEN returns of empty cells returns TRUE (just tested in Excel).

**Comment by nikazooz** [comment]
@nicolashefti I've made the changes

**Review by nicolashefti — APPROVED** [review]
